web/client/plugins/__tests__/Login-test.js

```diff
@@ -0,0 +1,131 @@
+
+describe('Login Plugin', () => {
+    const stateMocker = createStateMocker({ controls, security });
+    beforeEach((done) => {
+        document.body.innerHTML = '<div id="container"></div>';
+        setTimeout(done);
+    });
+
+    afterEach((done) => {
+        ReactDOM.unmountComponentAtNode(document.getElementById("container"));
+        document.body.innerHTML = '';
+        setTimeout(done);
+    });
+    describe('Login', () => {
+        it('default render', () => {
+            const storeState = stateMocker();
+            const { Plugin } = getPluginForTest(Login, storeState);
+            ReactDOM.render(<Plugin />, document.getElementById("container"));
+            expect(document.querySelector('#mapstore-login-menu')).toBeTruthy();
+            // permission table present by default
+            expect(document.querySelector('.modal-dialog')).toBeFalsy();
+        });
+        it('show login modal from external action', () => {
+            const storeState = stateMocker(toggleControl('LoginForm', 'enabled'));
+            const { Plugin } = getPluginForTest(Login, storeState);
+            ReactDOM.render(<Plugin />, document.getElementById("container"));
+            expect(document.querySelector('#mapstore-login-menu')).toBeTruthy();
+            // permission table present by default
+            expect(document.querySelector('.modal-dialog')).toBeTruthy();
+        });
+        it('show login modal on click', () => {
+            const storeState = stateMocker();
+            const { Plugin } = getPluginForTest(Login, storeState);
+            ReactDOM.render(<Plugin />, document.getElementById("container"));
+            TestUtils.Simulate.click(document.querySelector('.glyphicon-log-in'));
+            expect(document.querySelector('#mapstore-login-menu')).toBeTruthy();
+            // permission table present by default
+            expect(document.querySelector('.modal-dialog')).toBeTruthy();
+        });
+
+        it('render in OmniBar', () => {
+            const storeState = stateMocker();
+            // const { Plugin } = getPluginForTest(Login, storeState);
+            const { Plugin: OmniBarPlugin } = getPluginForTest(OmniBar, storeState, { LoginPlugin: Login });
+            ReactDOM.render(<OmniBarPlugin items={[{ ...Login.LoginPlugin.OmniBar, plugin: Plugin.LoginPlugin}]} />, document.getElementById("container"));
+            expect(document.querySelector('#mapstore-navbar-container .glyphicon-user')).toBeTruthy();
+        });
+        it('confirm dialog on pending changes', done => {
+            const storeState = stateMocker(
+                loginSuccess({  User: { name: "Test", access_token: "some-token"}}),
+                setControlProperty('unsavedMap', 'enabled', true),
+                setControlProperty('unsavedMap', 'source', 'logout')
+                // TODO: not sure about why this is required, investigate.
```

Comment:
what about this todo?

Reply:
Good catch. I removed that dependency at the end, but not the TODO :smile: 
```suggestion
```